underfs/ozone/src/main/java/alluxio/underfs/ozone/OzoneUnderFileSystemFactory.java

```diff
@@ -36,7 +36,7 @@ public boolean supportsPath(String path) {
   @Override
   public boolean supportsPath(String path, UnderFileSystemConfiguration conf) {
     if (supportsPath(path)) {
-      if (!conf.isSet(PropertyKey.UNDERFS_VERSION)
+      if (!conf.isSetByUser(PropertyKey.UNDERFS_VERSION)
```

Comment:
in this case, if default value of `PropertyKey.UNDERFS_VERSION` is removed in the future (not saying this will happen, but it could potentially).
this will make line 40 throw NullPointerException.

Perhaps this is the correct way to handle:

```suggestion
      if (!conf.isSet(PropertyKey.UNDERFS_VERSION) // treating this means matching all versions, 
          || (conf.isSetByUser(PropertyKey.UNDERFS_VERSION)) 
              && conf.get(PropertyKey.UNDERFS_VERSION).equals(getVersion()))
```

Reply:
@apc999 is the following way can avoid npe also？

```
getVersion().equals(conf.get(PropertyKey.UNDERFS_VERSION)))
```

Reply:
If the user does not set `PropertyKey.UNDERFS_VERSION`, then the value of `!conf.isSetByUser(PropertyKey.UNDERFS_VERSION)` is `true` (regardless of whether he has a default value or not), `||` will short-circuit and will not execute the following code, and the value of the expression of if will be `true`
If the user does set `PropertyKey.UNDERFS_VERSION`, no null pointer exception will occur